distribution.pm

```diff
@@ -139,13 +139,13 @@ sub script_run {
     if ($args{timeout} > 0) {
         my $str = testapi::hashed_string("SR" . $cmd . $args{timeout});
         if (testapi::is_serial_terminal) {
-            my $marker = " ; echo $str-\$?-";
+            my $marker = " ; echo $str-\$?- $args{output}";
             testapi::type_string($marker);
             testapi::wait_serial($cmd . $marker, no_regex => 1, quiet => $args{quiet});
             testapi::type_string("\n");
         }
         else {
-            testapi::type_string " ; echo $str-\$?- > /dev/$testapi::serialdev\n";
+            testapi::type_string " ; echo $str-\$?- $args{output} > /dev/$testapi::serialdev\n";
```

Comment:
as you add it twice please extract the generated string part into a variable. Actually `$marker` from L142 seems to be a good candidate already so probably you can simply do:

```perl
my $marker = " ; echo $str-\$?- $args{output}";
if (testapi::is_serial_terminal) {
    testapi::type_string($marker);
    …
}
else {
    testapi::type_string "$marker > /dev/$testapi::serialdev\n";
}
```

or just combine it all in `$str` already.